### DISCUSSION TITLE
Different weather icons for different precipitation intensity

### DISCUSSION THREAD

**Discussion by ccyybn** [breezy-weather/breezy-weather#1321] [Ideas]
### Describe your idea

![Snipaste_2024-09-21_23-51-59](https://github.com/user-attachments/assets/d8a4c9de-5079-4988-9e36-dce9f89642b8)

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.8 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
Not all weather sources support it: you suggest 4 levels of intensity but some only have one icon, others would have 2 or 3 or 4, so it would be hard to make it match all of our sources.

Additionally, we support many icon packs. How would we maintain the compatibility with icon packs with only one icon?

The precipitation quantity (when supported by the source) remains the most reliable IMO.

Is your screenshot an actual implementation or is it a mockup?

**Reply by ccyybn** [reply]
It is an actual implementation. I also modified the `flat icon pack` to provide icons for different intensities. If the icon pack does not have a rain icon for a specific intensity, it will now fall back to the default rain icon in the app.

I think that if the icon pack was made previously and does not offer icons of different intensities, the app can just use that single icon regardless of the rainfall intensity, and then people using the icon pack wouldn't notice the change.

Actually the Chinese source has 6 intensity levels, but I mapped the top 3 to the same icon (because the top 2 are extremely rare), so if a source only has 2 or 3, it can use only 2 or 3 of them.
[drawable.zip](https://github.com/user-attachments/files/17085294/drawable.zip)
